chipsec/helper/linuxnative/legacy_pci.py

```diff
@@ -57,49 +59,54 @@
     0x90,                   # nop
     0x5d,                   # pop    %rbp
     0xc3,                   # retq
-]
+))
 
 
-class PORTS:
-    def __init__(self):
-        clib = cdll.LoadLibrary("libc.so.6")
-        clib.iopl(3)
+class Ports:
+    # Use a unique Ports instance, to avoid allocating memory mappings every time it is used
+    instance = None
 
-        opc = IN_PORT
-        size = len(opc)
-        code = (c_ubyte * size)(*opc)
-        self.addr = mmap.mmap(-1, mmap.PAGESIZE, prot=mmap.PROT_READ | mmap.PROT_WRITE | mmap.PROT_EXEC)
-        self.addr.write(bytes(code))
+    def __init__(self) -> None:
+        clib = CDLL("libc.so.6", use_errno=True)
+        if clib.iopl(3) == -1:
+            raise OsHelperError("Unable to use I/O ports using iopl", get_errno())
+
+        self.inl_addr = mmap.mmap(-1, mmap.PAGESIZE, flags=mmap.MAP_PRIVATE, prot=mmap.PROT_READ | mmap.PROT_WRITE | mmap.PROT_EXEC)
```

Comment:
Why is MAP_PRIVATE needed?

Reply:
By default, `mmap.mmap` creates a `mmap.MAP_SHARED` mapping. In this function, there is no need to have a shared mapping, which is why I added `MAP_PRIVATE`.

Also, while debugging other issues with `/proc/self/maps` I stumbled upon this mapping, described there as `/dev/zero (deleted)`. As this is uncommon, I wanted to change it to `MAP_PRIVATE` and did not find any regression while testing.